transcom/mymove
# B-21685 INT - TOO can create shipment even when no shipments exist on move

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21685)

## Summary

This BL is to let the TOO add a shipment even when no shipments are on the move (such as the shipments being deleted). The SC could already do that, and I made the TOO match the SC as much as I could while also tidying things up for both.

- Gave TOO capability to add a new shipment even when no requested or approved shipments exist on the move
- Pulled the SC's "Flag move for financial review" button out of the shipment components and up higher on the page (to match the TOO's view), since it's at the move level and not shipment level

The SC and TOO have slightly different components, as the TOO has a section for Requested and another section for Accepted shipments. The SC only has one section.

In the case of a TOO deleting all the shipments on a move, there would be no way to add new shipments since the Add a new shipment button is tied into both the Accepted and Requested shipment components.

I thought about taking the add a new shipment button out of both of those components and just having it up at the top, but instead went with keeping the existing functionality but then having a separate section just labeled the basic "Shipment" with an Add a new shipment button in the case of neither Accepted or Requested shipments. In the case of no shipments on a move, the SC and TOO will see the same section and button now, and should be able to add a new shipment:
![image](https://github.com/user-attachments/assets/b92b039a-298d-4824-8ee2-045a06043a24)


I did make the SC and TOO match as best I could, and pulled the SC's "Flag move for financial review" button out of the shipment components, since it's at the move level and not shipment level.

### How to test

1. Access the app as a customer
2. Create a move with multiple shipments
3. View as both SC and TOO and make sure it looks normal
4. Delete both shipments by clicking on each shipment and clicking "Delete Shipment" towards the top of the page
5. Verify you can add a shipment as the TOO now

Adding UB Shipments
1. Have the unaccompanied_baggage FF set to on
2. Access the app as a customer
3. Create a move with one or both duty locations being OCONUS
4. You should be able to create a UB shipment as a customer. Up to you on whether you add one now as a customer (we will be testing adding as an office user)
5. Complete the move
6. As the SC, verify that you can see UB as a shipment type to add after clicking on the Add a new shipment button
![image](https://github.com/user-attachments/assets/dadf6de0-af03-4220-a64d-6340ffb6205e)
7. As the TOO, verify that you can also see UB as a shipment type to add after clicking on the Add a new shipment button
![image](https://github.com/user-attachments/assets/489be632-86c5-4d2b-a932-3a27a4600cfe)
8. Delete the shipments from the move
9. As the SC, verify that you can still see UB as a shipment type to add after clicking on the Add a new shipment button
![image](https://github.com/user-attachments/assets/75d19c50-f1bb-419e-abcb-41b657f468d9)
10. As the TOO, verify that you can still see UB as a shipment type to add after clicking on the Add a new shipment button
![image](https://github.com/user-attachments/assets/efd16c22-d3d3-4f2a-a587-6794ac36d2c6)
11. Set the unaccompanied_baggage FF to off, rerun direnv allow and rerun the server, refresh the page
12. Verify that you do not see UB in the drop down list for the SC to add a shipment
![image](https://github.com/user-attachments/assets/455f595e-cef8-48d5-bcd8-69113f30a8b5)
13. Verify that you do not see UB in the drop down list for the TOO to add a shipment
![image](https://github.com/user-attachments/assets/ef471215-8cc0-42cd-b5d9-37ffaffe6778)


![image](https://github.com/user-attachments/assets/5a17d032-fda9-48db-b8dc-3ca9624c3a12)

![image](https://github.com/user-attachments/assets/1badeaad-9122-4b43-81cc-4611a60096c4)


## Screenshots
SC (left) vs TOO (right) view for multiple shipments submitted:
- only new thing here is the move of Flag Move for financial review up to the move details on the left, SC view:
![image](https://github.com/user-attachments/assets/38e3a07b-965e-46b3-8ee9-2793081db5c2)

SC (left) vs TOO (right) view for all shipments deleted from a move:
- New on the left: the move of Flag Move for financial review up to the move details on the left, SC view
- New on the right for the TOO: the Add a new shipment button in the shipments section, which matches the SC view
![image](https://github.com/user-attachments/assets/e3480a7d-4cb7-46f0-b560-656cc39e0ba1)
![image](https://github.com/user-attachments/assets/ecae011a-debd-455d-b3b1-bd180abae11f)
